src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/retrycontext/CallbackRetryContextBuilder.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.ccd.domain.service.callbacks.retrycontext;
+
+import com.google.common.collect.Lists;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.ApplicationParams;
+
+import java.util.List;
+
+@Service
+public class CallbackRetryContextBuilder {
+    public static final int CALLBACK_RETRY_INTERVAL_MULTIPLIER = 3;
+    private final List<Integer> defaultCallbackRetryIntervalsInSeconds;
+    private final Integer defaultCallbackTimeoutInSeconds;
+
+    public CallbackRetryContextBuilder(final ApplicationParams applicationParams) {
+        this.defaultCallbackRetryIntervalsInSeconds = applicationParams.getCallbackRetryIntervalsInSeconds();
+        this.defaultCallbackTimeoutInSeconds = applicationParams.getCallbackTimeoutInSeconds();
+    }
+
+    public List<CallbackRetryContext> buildCallbackRetryContexts(final List<Integer> callbackRetryTimeouts) {
+        List<CallbackRetryContext> retryContextList = Lists.newArrayList();
+        if (callbackRetryTimeouts.isEmpty() || isCallbackRetriesWithZeros(callbackRetryTimeouts)) {
+            buildDefaultRetryContext(retryContextList);
+        } else if (isCallbackRetriesDisabled(callbackRetryTimeouts)) {
+            disableRetryContext(retryContextList);
+        } else {
+            buildCustomRetryContext(callbackRetryTimeouts, retryContextList);
+        }
+        return retryContextList;
+    }
+
+    private boolean isCallbackRetriesWithZeros(final List<Integer> callbackRetryTimeouts) {
+        return callbackRetryTimeouts.size() != 1 && callbackRetryTimeouts.stream().anyMatch(callback -> callback == 0);
+    }
+
+    private void disableRetryContext(final List<CallbackRetryContext> retryContextList) {
+        retryContextList.add(new CallbackRetryContext(0, defaultCallbackTimeoutInSeconds));
+    }
+
+    private void buildDefaultRetryContext(final List<CallbackRetryContext> retryContextList) {
+        this.defaultCallbackRetryIntervalsInSeconds.forEach(cbRetryInterval -> retryContextList.add(
+            new CallbackRetryContext(cbRetryInterval, defaultCallbackTimeoutInSeconds)));
+    }
+
+    private void buildCustomRetryContext(final List<Integer> callbackRetryTimeouts, final List<CallbackRetryContext> retryContextList) {
+        retryContextList.add(new CallbackRetryContext(0, callbackRetryTimeouts.remove(0)));
```

Comment:
here and in line 48 we are hardcoding the values 0 and 1. This issue with this is that if we change the config `ccd.callback.retry.intervals=0,1,3` that is not picked up here and could have some misleading behaviour

Reply:
The `ccd.callback.retry.intervals` is only used for building default retry context and this is custom retry context. It works on the notion we multiple every successive interval value by 3. Do you mean we should provide a multiplier in the properties as well? That was not the requirement of the ticket however.

Reply:
I meant something different. According to the ticket, the retry interval should always be 0,1,3... for both default and custom callback settings.
The way things are implemented now it's possible instead that we can end up with different values for the retry interval used for the default behaviour and the custom behaviour:
Example: 
change ccd.callback.retry.intervals to say 2,3,10. Now default callbacks keep using 0,1,3 but the custom start using 2,3,10. So there is a divergence which makes no sense.
So we either remove 'ccd.callback.retry.intervals', so that it cannot be reconfigured and it's always hardcoded to 0,1,3. Or we change the code below to pick the new values for 'ccd.callback.retry.intervals'.
But we cannot let the 2 diverge.

Reply:
Yes ok I will change to keep the default retry intervals property and use it in the first 3 tries of custom context.

Reply:
done